packages/core/test/chainbridge/sinkOracle.js

```diff
@@ -0,0 +1,149 @@
+        event.ancillaryData.toLowerCase() === testAncillary.toLowerCase()
+    );
+    assert.isFalse(
+      await sinkOracle.hasPrice(testIdentifier, testRequestTime, testAncillary, { from: owner }),
+      "should not have price after request"
+    );
+    assert(
+      await didContractThrow(sinkOracle.getPrice(testIdentifier, testRequestTime, testAncillary, { from: owner })),
+      "should revert after request price"
+    );
+    // Deposit event will be emitted after successful Bridge.deposit() internal call if the resource ID is set up
+    // properly.
+    const internalTxn = await TruffleAssert.createTransactionResult(bridge, txn.tx);
+    TruffleAssert.eventEmitted(
+      internalTxn,
+      "Deposit",
+      event =>
+        event.destinationChainID.toString() === destinationChainID.toString() &&
+        event.resourceID.toLowerCase() === sinkOracleResourceId.toLowerCase() &&
+        event.depositNonce.toString() === expectedDepositNonce.toString()
+    );
+  });
+  it("validateDeposit", async function() {
+    assert(
+      await didContractThrow(sinkOracle.validateDeposit(chainID, testIdentifier, testRequestTime, testAncillary)),
+      "Reverts if price not requested yet"
+    );
+    await sinkOracle.requestPrice(testIdentifier, testRequestTime, testAncillary, { from: owner });
+    await sinkOracle.validateDeposit(chainID, testIdentifier, testRequestTime, testAncillary);
+  });
+  it("executePublishPrice", async function() {
+    await sinkOracle.requestPrice(testIdentifier, testRequestTime, testAncillary, { from: owner });
+    assert(
+      await didContractThrow(
+        sinkOracle.executePublishPrice(chainID, testIdentifier, testRequestTime, testAncillary, { from: rando })
+      ),
+      "Only callable by GenericHandler"
+    );
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.GenericHandler), rando);
+    await sinkOracle.executePublishPrice(chainID, testIdentifier, testRequestTime, testAncillary, testPrice, {
+      from: rando
+    });
+    assert(
+      await didContractThrow(sinkOracle.hasPrice(testIdentifier, testRequestTime, testAncillary, { from: rando })),
+      "should revert if not called by registered contract"
+    );
+    assert(
+      await didContractThrow(sinkOracle.getPrice(testIdentifier, testRequestTime, testAncillary, { from: rando })),
+      "should revert if not called by registered contract"
+    );
+    assert.isTrue(
+      await sinkOracle.hasPrice(testIdentifier, testRequestTime, testAncillary, { from: owner }),
+      "should have price after publish"
+    );
+    assert.equal(
+      (await sinkOracle.getPrice(testIdentifier, testRequestTime, testAncillary, { from: owner })).toString(),
+      testPrice,
+      "should not revert after publish"
+    );
+  });
+  it("formatMetadata", async function() {
+    const metadata = await sinkOracle.formatMetadata(chainID, testIdentifier, testRequestTime, testAncillary);
```

Comment:
Just curious, is this method only public so we can test it? Or is it needed for some other reason?

Reply:
Yeah its public primarily for testing, but its also convenient in case we need to call Bridge.deposit() directly, because the `metadata` return value can be passed directly as the third param to `deposit()`